core/src/main/java/edu/wpi/grip/core/metrics/Timer.java

```diff
@@ -0,0 +1,131 @@
+package edu.wpi.grip.core.metrics;
+
+import edu.wpi.grip.core.events.TimerEvent;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Stopwatch;
+import com.google.common.eventbus.EventBus;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Timer for code that gets run.
+ *
+ * <p>Sample usage:
+ * <pre><code>
+ *   Timer myTimer = ...
+ *   myTimer.time(this::doSomething);
+ * </code></pre>
+ * </p>
+ */
+public class Timer {
+
+  private final EventBus eventBus;
+  private final Stopwatch stopwatch;
+  private boolean started = false;
```

Comment:
private variable started can be replaced with a call to stopwatch.isRunning()

https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Stopwatch.java